### PR TITLE
feat(@computedFrom): report invalid references from `@computedFrom` d…

### LINKED ISSUE #128: report invalid references from `@computedFrom`
Following

``` TypeScript
      export class Foo{
        field1: number;
        field2: number;

        @computedFrom("field1", "missingField1", "field2", "missingField2")
        get computedField(){
          return this.field1 + this.field2;
        }
      }
```

``` HTML
      <template>
        \${computedField}
      </template>
```

should report that missingField1 and missingField2 can't be found in type Foo

### PR DESCRIPTION
**Pull request by atsu85**
Make sure you check the following: 
* [x] you are making a pull request for the `develop` branch
* [x] unless trivial, a corresponding issue exists for this PR (reference it)
* [x] if this PR fixes the issue, then include `fix #`
* [x] you've ran `gulp test` and it passes the lint
* [x] tests don't fail (except tests in 'failing.spec.ts')

***Note:** Requests to pull into master from a fork will be rejected.*
…ecorator

fix #128

### CODE REVIEW

**Comment by MeirionHughes** [comment]
Looks good. published.

File: source/rules/binding.ts [hunk 1]
```diff
@@ -475,6 +482,28 @@ export class BindingRule extends ASTBuilder {
...
   }
 
+  private checkDecorators(member, context: ASTContext, loc: FileLoc) {
+    if (member.decorators) {
+      const memberNode = <Node>member;
+      const decorators: NodeArray<Decorator> = member.decorators;
+      decorators.forEach((decorator) => {
+        const decoratorExprNode = <any>decorator.expression;
+        const expr = <Identifier>decoratorExprNode.expression;
+        if (expr.text === "computedFrom") {
```
Comment (atsu85) [inline source/rules/binding.ts]: @MeirionHughes, I'm not sure, if I should have implemented smarter check for `@computedFrom` decorator (imported from "aurelia-binding") that potentially could be renamed when importing it (don't know why anyone should do it, but it is possible). I guess I could inspect imports, but probably there could be problem when the decorator is imported from another module, that reexports this decorator. That's why I currently just check decorator name. What do You think about it?

File: source/rules/binding.ts [hunk 2]
```diff
@@ -475,6 +482,28 @@ export class BindingRule extends ASTBuilder {
...
+          var decoratorArguments = decoratorExprNode.arguments;
+          const decoratorArgumentsAsText: string[] = decoratorArguments.map((decoratorArg) => decoratorArg.text);
+          decoratorArgumentsAsText.forEach((computedDependencyText) => {
+            if (computedDependencyText.indexOf(".") === -1) {
+              this.resolveStaticType(context, computedDependencyText, loc);
+            } else {
+              console.warn("TODO: Checking references from nested object (" + computedDependencyText + ") is not yet implemented for @computedFrom decorator!");
```
Comment (atsu85) [inline source/rules/binding.ts]: @MeirionHughes,
I may have misunderstood Your [comment](https://github.com/MeirionHughes/aurelia-template-lint/issues/128#issuecomment-252362174)
> you could wrap the string with ${_expression_}

but when I replaced this line with
`this.resolveStaticType(context, "${" + computedDependencyText + "}", loc);`

it didn't work either (probably I should pass `"${" + computedDependencyText + "}"` to smth that creates expression?). Any hints?

Comment (atsu85) [inline source/rules/binding.ts]: also `this.createAttributeInstruction("none", "bind", computedDependencyText, loc.line, loc.column);` doesn't work :/

**Comment by MeirionHughes** [comment]
I've done it. Needed a little refactoring, but it should work okay for now.

File: source/rules/binding.ts (hunk 1 above)
Comment (MeirionHughes) [inline source/rules/binding.ts]: Its possible I guess; and there are quite a few ways to break it too if you really wanted to. Worrying about them all is not worth the effort at this stage.

Comment (atsu85) [inline source/rules/binding.ts]: > Worrying about them all is not worth the effort at this stage.

I agree, just wanted to hear Your opinion as well :)

**Pull request merged**